lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -268,3 +280,90 @@ instance CompareDiscovery (RndState n) where
 
 instance KnownAddresses (RndState n) where
     knownAddresses s = Map.elems (addresses s)
+
+--------------------------------------------------------------------------------
+--
+-- RndAnyState
+--
+-- For benchmarking and testing arbitrary large random wallets.
+
+-- | An "unsound" alternative that can be used for benchmarking and stress
+-- testing. It re-uses the same underlying structure as the `RndState` but
+-- it discover addresses based on an arbitrary ratio instead of decrypting the
+-- derivation path.
+--
+-- The proportion is stored as a type-level parameter so that we don't have to
+-- alter the database schema to store it. It simply exists and depends on the
+-- caller creating the wallet to define it.
+newtype RndAnyState (network :: NetworkDiscriminant) (p :: Nat) = RndAnyState
+    { innerState :: RndState network
+    } deriving (Generic, Show)
+
+instance NFData (RndAnyState n p)
+
+-- | Initialize the HD random address discovery state from a root key and RNG
+-- seed.
+--
+-- The first argument is expected to be a ratio (between 0 and 1) of addresses
+-- we ought to simply recognize as ours. So, giving .5 means that 50% of the
```

Comment:
Seems outdated. Should be `ratio (between 0% and 100%)` and `giving @50 means 50% of the` I presume?

Reply:
Right, I had it as an argument initially and switched to a type-level parameter later on. 

> giving @50 means 50% of the I presume?

Indeed.